### PR TITLE
fix: resource/project_ip_whitelist - modify ip whitelist entry valida…

### LINKED ISSUE #252: Adding 16 whitelist entries at the same time causes an error
Steps to Reproduce
1. Create a new project. 
2. Create a .tf file to add 16 whitelist entries. I have used the below:

`provider "mongodbatlas" {
  public_key = "xxxx" 
  private_key  = "xxxx"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist1" {
project_id = "xxxx"
cidr_block = "10.0.0.0/8"
comment = "whitelist CIDR block1"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist2" {
project_id = "xxxx"
cidr_block = "130.99.0.0/16"
comment = "whitelist CIDR block2"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist3" {
project_id = "xxxx"
cidr_block = "136.229.0.0/16"
comment = "whitelist CIDR block3"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist4" {
project_id = "xxxx"
cidr_block = "143.122.0.0/16"
comment = "whitelist CIDR block4"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist5" {
project_id = "xxxx"
cidr_block = "165.152.0.0/16"
comment = "whitelist CIDR block5"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist6" {
project_id = "xxxx"
cidr_block = "169.4.0.0/14"
comment = "whitelist CIDR block6"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist7" {
project_id = "xxxx"
cidr_block = "169.8.0.0/13"
comment = "whitelist CIDR block7"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist8" {
project_id = "xxxx"
cidr_block = "172.16.0.0/12"
comment = "whitelist CIDR block8"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist9" {
project_id = "xxxx"
cidr_block = "192.28.0.0/18"
comment = "whitelist CIDR block9"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist10" {
project_id = "xxxx"
cidr_block = "192.28.64.0/19"
comment = "whitelist CIDR block10"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist11" {
project_id = "xxxx"
cidr_block = "192.28.96.0/22"
comment = "whitelist CIDR block11"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist12" {
project_id = "xxxx"
cidr_block = "192.43.235.0/24"
comment = "whitelist CIDR block12"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist13" {
project_id = "xxxx"
cidr_block = "202.237.244.0/24"
comment = "whitelist CIDR block13"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist14" {
project_id = "xxxx"
cidr_block = "204.126.212.0/23"
comment = "3M whitelist CIDR block14"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist15" {
project_id = "xxxx"
cidr_block = "204.221.11.0/24"
comment = "whitelist CIDR block15"
}

resource "mongodbatlas_project_ip_whitelist" "cckm_project_whitelist16" {
project_id = "xxxx"
cidr_block = "205.153.108.0/22"
comment = "whitelist CIDR block16"
}
`

3. terraform init-validate-appply

After completing the above, some IP whitelist entries are created, but the process fails with the output below from Terraform:

`
Error: Provider produced inconsistent result after apply
When applying changes to
mongodbatlas_project_ip_whitelist.cckm_project_whitelist8, provider
"registry.terraform.io/-/mongodbatlas" produced an unexpected new value for
was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.
`

If you retry terraform apply, the remaining IP whitelist entries are created successfully.

I have used * provider.mongodbatlas: version = "~> 0.6" and Terraform v0.12.24 in my test.

### PR DESCRIPTION
**Pull request by marinsalinas**
…tion and fix acctest

## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.
Closes #252 

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the MongoDB CLA
- [x] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
```
--- PASS: TestAccResourceMongoDBAtlasProjectIPWhitelist_SettingMultiple (139.76s)
PASS
coverage: 4.7% of statements
ok      github.com/terraform-providers/terraform-provider-mongodbatlas/mongodbatlas     (cached)        coverage: 4.7% of statements
```

### CODE REVIEW

**Review by PacoDw — APPROVED** [review]
LGMT!

**Review by coderGo93 — APPROVED** [review]
LGTM, only needs to be fixed for travis check

File: mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go [hunk 1]
```diff
@@ -178,8 +185,13 @@ func resourceMongoDBAtlasProjectIPWhitelistRead(d *schema.ResourceData, meta int
 			case strings.Contains(fmt.Sprint(err), "500"):
 				return resource.RetryableError(err)
 			case strings.Contains(fmt.Sprint(err), "404"):
-				d.SetId("")
-				return nil
+				log.Printf("[DEBUG] whitelist entry %s is not found", ids["entry"])
```
Comment (coderGo93) [inline mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go]: Is necessary this line?

File: mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go [hunk 2]
```diff
@@ -127,8 +127,15 @@ func resourceMongoDBAtlasProjectIPWhitelistCreate(d *schema.ResourceData, meta i
...
+				whiteListEntry := ipAddress
+				if cidrBlock != "" {
+					whiteListEntry = cidrBlock
+				}
+
 				for _, entry := range whitelist {
-					if entry.IPAddress == ipAddress || entry.CIDRBlock == cirdBlock {
+					if entry.IPAddress == whiteListEntry || entry.CIDRBlock == whiteListEntry {
+
+						log.Printf("[DEBUG] Entry %+v is created, ipAddress=%s, cidrBlock=%s", entry, ipAddress, cidrBlock)
```
Comment (coderGo93) [inline mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go]: Is necessary this line to keep it?

**Review by themantissa — APPROVED** [review]
lgtm - thank your for staying on this one

**Pull request merged**